### ISSUE TITLE
URI encode jsonGraph payload

### ISSUE THREAD

**Issue by ekosz** [Netflix/falcor-http-datasource#31]
Given I run the command:

```js
falcor.setValue(['somethingById', 'abc123', 'name'], 'This & That').then(something);
```

The JSON payload that is set to the server looks like:

<img width="995" alt="screen shot 2016-01-27 at 2 49 30 pm" src="https://cloud.githubusercontent.com/assets/212829/12631366/bfe897e0-c505-11e5-86b3-699b67594ff4.png">

And an error is caused on the backend:

```
 SyntaxError: Unexpected end of input
     at Object.parse (native)
     at /app/node_modules/falcor-express/src/requestToContext.js:21:37
     at Array.forEach (native)
     at requestToContext (/app/node_modules/falcor-express/src/requestToContext.js:18:31
     at /app/node_modules/falcor-express/src/index.js:14:23
     ...
```

It seems that falcor should URI encode its jsonGraph payload before posting.

This was transplanted from https://github.com/Netflix/falcor/issues/722

**Comment by ekosz** [comment]
This seems like it may also relate to #29 ?

**Comment by ekosz** [comment]
To add onto this, I've found just uri encoding before sending will also break the the application when there are new lines present.

**Comment by ekosz** [comment]
I keep finding holes in rolling my own solution, does anyone at the Netflix team know a good way to solve this kind of issue?

**Comment by trxcllnt** [comment]
@ekosz taking a shot in the dark here, but have you tried using `encodeURI` on the entire payload rather than `encodeURIComponent` on individual paths?

**Comment by ekosz** [comment]
Hey @trxcllnt do you know if there is a way to do that while still using `falcor.set`?

**Comment by trxcllnt** [comment]
@ekosz I think you'd need to modify [XMLHttpSource#set](https://github.com/Netflix/falcor-http-datasource/blob/master/src/XMLHttpSource.js#L56) to [build](https://github.com/Netflix/falcor-http-datasource/blob/master/src/buildQueryObject.js) the query Object differently. Can you try monkey-patching XMLHttpSource.prototype.set with an implementation that calls `encodeURI` instead of reusing the `buildQueryObject` function?

**Comment by ekosz** [comment]
Looks like I was just using an old version and didn't realize it. This should have been fixed in version `0.1.3`.
